test/e2e/cluster_upgrade_runtimesdk.go

```diff
@@ -478,9 +496,33 @@ func runtimeHookTestHandler(ctx context.Context, c client.Client, namespace, clu
 		fmt.Sprintf("ClusterTopology reconcile did not unblock after updating hook response: hook %s still blocking", hookName))
 }
 
-// clusterConditionShowsHookFailed checks if the TopologyReconciled condition message contains both the hook name and hookFailedMessage.
-func clusterConditionShowsHookFailed(cluster *clusterv1.Cluster, hookName string) bool {
-	return conditions.GetReason(cluster, clusterv1.TopologyReconciledCondition) == clusterv1.TopologyReconcileFailedReason &&
-		strings.Contains(conditions.GetMessage(cluster, clusterv1.TopologyReconciledCondition), hookFailedMessage) &&
+// clusterConditionShowsHookBlocking checks if the TopologyReconciled condition message contains both the hook name and hookFailedMessage.
+func clusterConditionShowsHookBlocking(cluster *clusterv1.Cluster, hookName string) bool {
+	return conditions.GetReason(cluster, clusterv1.TopologyReconciledCondition) == clusterv1.TopologyReconciledHookBlockingReason &&
 		strings.Contains(conditions.GetMessage(cluster, clusterv1.TopologyReconciledCondition), hookName)
 }
+
+func backupAndDeleteCluster(ctx context.Context, proxy framework.ClusterProxy, namespace, clusterName, artifactFolder string) {
+	By("Deleting the workload cluster")
+	cluster := &clusterv1.Cluster{}
+	Eventually(func() error {
+		return proxy.GetClient().Get(ctx, client.ObjectKey{Namespace: namespace, Name: clusterName}, cluster)
+	}).Should(Succeed())
+
+	// Dump all the logs from the workload cluster before deleting them.
+	proxy.CollectWorkloadClusterLogs(ctx, cluster.Namespace, cluster.Name, filepath.Join(artifactFolder, "clusters-beforeClusterDelete", cluster.Name))
+
+	// Dump all Cluster API related resources to artifacts before deleting them.
+	framework.DumpAllResources(ctx, framework.DumpAllResourcesInput{
+		Lister:    proxy.GetClient(),
+		Namespace: namespace,
+		LogPath:   filepath.Join(artifactFolder, "clusters-beforeClusterDelete", proxy.GetName(), "resources"),
+	})
+
+	By("Deleting the workload cluster")
+	// Delete the workload cluster and call the before ClusterDeleteTestHandler.
+	framework.DeleteCluster(ctx, framework.DeleteClusterInput{
```

Comment:
Not relevant to this PR but just FYI incase you want t fix it: The comment on `framework.DeleteCluster` is misleading. The current comment suggests that it waits for cluster owned objects to be gone.